src/Ledger/Utxo.lagda

```diff
@@ -485,32 +495,45 @@ data
 \begin{code}[hide]
 data _⊢_⇀⦇_,UTXOS⦈_ where
 \end{code}
+\begin{AgdaMultiCode}
 \begin{code}
   Scripts-Yes :
-    ∀ {Γ} {s} {tx}
-    → let open Tx tx renaming (body to txb); open TxBody txb
-          open UTxOEnv Γ renaming (pparams to pp)
-          open UTxOState s
-          sLst = collectPhaseTwoScriptInputs pp tx utxo
+    let  pp         = Γ .pparams
+         utxo       = s .utxo
+         fees       = s .fees
+         deposits   = s .deposits
+         donations  = s .donations
```

Comment:
I think this would be nicer as a fake pattern-match below. It has the added benefit that the fields of the state would line up nicely.

Reply:
What do you mean by fake pattern-match?

Reply:
I mean to construct `s` as an index in the conclusion, like with the final state, i.e.
```
Γ ⊢ ⟦ utxo , fees , deposits , donations ⟧ ⇀⦇ tx ,UTXOS⦈ ⟦ (utxo ∣ txins ᶜ) ∪ˡ (outs txb) , fees + txfee , updateDeposits pp txb deposits , donations + txdonation ⟧
```
I call it fake pattern match, because it looks and feels like one (as if pattern-matching on `s`) but it isn't.

Reply:
That looks reasonable to me.  Actually, both options look good to me, but I think what Andre suggests is more consistent with the other step rules.